cmake/onnxruntime_java.cmake

```diff
@@ -134,6 +134,10 @@ if (WIN32)
   if(NOT onnxruntime_ENABLE_STATIC_ANALYSIS)
     add_custom_command(TARGET onnxruntime4j_jni POST_BUILD COMMAND ${CMAKE_COMMAND} -E create_symlink $<TARGET_FILE:onnxruntime> ${JAVA_PACKAGE_LIB_DIR}/$<TARGET_FILE_NAME:onnxruntime>)
     add_custom_command(TARGET onnxruntime4j_jni POST_BUILD COMMAND ${CMAKE_COMMAND} -E create_symlink $<TARGET_FILE:onnxruntime4j_jni> ${JAVA_PACKAGE_JNI_DIR}/$<TARGET_FILE_NAME:onnxruntime4j_jni>)
+    if (onnxruntime_USE_CUDA)
```

Comment:
This block is only in the WIN32 branch, and needs to be in the else case as well, with the appropriate substitution of `TARGET_FILE_NAME` for `TARGET_LINKER_FILE_NAME`. Without it it doesn't bundle the cuda and shared provider libraries into the jar on Linux.

Reply:
Ah, I see it. It's unfortunate that these have to be duplicated with a slight change for multiple platforms.

Reply:
I'm going to have to refactor the Java cmake bits a little anyway to add in support for the other shared library providers, so provided there is time to do it before this code is released we're ok.

Reply:
It might be that I don't understand how CMake works well enough to get around it, but when I looked into it last year this was the simplest way I could figure out.

Reply:
No worries, I pushed a change just a minute ago with what you said and made sure to do the substitution. Cmake is certainly nebulous... and not easy to know when you did something wrong.